### ISSUE TITLE
configs with missing data mess up printed energy errors (but not loss?)

### ISSUE THREAD

**Issue by bernstei** [ACEsuit/mace#1292]
I'm on commit 075c5444fd, `develop` branch as of Nov 20 or so.

I'm doing iterative training, and in one iteration I get models with fitting output like

```
2025-10-31 16:24:14.526 INFO: Epoch 42: head: pt_head, loss=0.00003488, RMSE_E_per_atom=   20.43 meV, RMSE_F=  216.14 meV / A, RMSE_stress=    5.25 meV / A^3
2025-10-31 16:24:14.656 INFO: Epoch 42: head: Default, loss=0.00000737, RMSE_E_per_atom=    1.83 meV, RMSE_F=   15.51 meV / A, RMSE_stress=    1.29 meV / A^3
.
.
.
+---------------+---------------------+------------------+-------------------+---------------------------------------+
|  config_type  | RMSE E / meV / atom | RMSE F / meV / A | relative F RMSE % | RMSE Stress (Virials) / meV / A (A^3) |
+---------------+---------------------+------------------+-------------------+---------------------------------------+
| train_Default |            1.9      |         13.0     |          2.24     |                     1.3               |
| valid_Default |            2.0      |         13.5     |          2.23     |                     1.2               |
+---------------+---------------------+------------------+-------------------+---------------------------------------+
```

The following fit I get
```
2025-11-03 16:17:03.285 INFO: Epoch 42: head: pt_head, loss=0.00003565, RMSE_E_per_atom=   19.06 meV, RMSE_F=   80.89 meV / A, RMSE_stress=    3.40 meV / A^3
2025-11-03 16:17:03.435 INFO: Epoch 42: head: Default, loss=0.00000769, RMSE_E_per_atom=  460.40 meV, RMSE_F=   16.05 meV / A, RMSE_stress=    1.36 meV / A^3
.
.
.
+---------------+---------------------+------------------+-------------------+---------------------------------------+ 
|  config_type  | RMSE E / meV / atom | RMSE F / meV / A | relative F RMSE % | RMSE Stress (Virials) / meV / A (A^3) |
+---------------+---------------------+------------------+-------------------+---------------------------------------+
| train_Default |          233.1      |         13.9     |          2.25     |                     1.3               |
| valid_Default |          460.2      |         14.1     |          2.24     |                     1.3               |
+---------------+---------------------+------------------+-------------------+---------------------------------------+
```

As you can see, the reported energy error is now crazy high, but everything else (force and stress errors, and even the _loss_) is quite reasonable.  I suspect that there are configurations with no `REF_*` fields, but I'll work on a minimal example.

**Comment by bernstei** [comment]
Some data for reproducing.  A small fitting data file, and two different validation files.  In the "normal" one all 20 configs have REF_ data, and in the "missing" one there are 20 additional configs with no reference data.

The following fits
```
mace_run_train --name=normal_normal --train_file=configs_fit_normal.extxyz --valid_file=configs_valid_normal.extxyz --foundation_model=small --max_num_epochs 25 --multiheads_finetuning=false
mace_run_train --name=normal_missing --train_file=configs_fit_normal.extxyz --valid_file=configs_valid_missing.extxyz --foundation_model=small --max_num_epochs 25 --multiheads_finetuning=false
```
produce very weird output.  The "normal" one ends with 
```
2025-11-26 11:01:06.878 INFO: Epoch 24: head: Default, loss=0.00484186, RMSE_E_per_atom=   35.63 meV, RMSE_F=   21.71 meV / A
```
while the "missing" one ends with
```
2025-11-26 10:59:58.545 INFO: Epoch 24: head: Default, loss=0.00242093, RMSE_E_per_atom=   77.85 meV, RMSE_F=   21.71 meV / A
```
which has identical force error, larger energy error, but _lower_ loss.  My suspicion is that in the reported energy error the ones without data are not being excluded, and instead being averaged in as zeros.

[configs_fit_normal.extxyz.txt](https://github.com/user-attachments/files/23775543/configs_fit_normal.extxyz.txt)
[configs_valid_missing_data.extxyz.txt](https://github.com/user-attachments/files/23775542/configs_valid_missing_data.extxyz.txt)
[configs_valid_normal.extxyz.txt](https://github.com/user-attachments/files/23775544/configs_valid_normal.extxyz.txt)

**Comment by bernstei** [comment]
Looks like 69afa61b1a2e removed some (necessary) calls to `filter_nonzero_weight`.  I'll try to fix that in a PR
